packages/test-base/src/androidTest/kotlin/io/realm/kotlin/test/shared/notifications/RealmListQueryTests.kt

```diff
@@ -0,0 +1,250 @@
+        val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+        val result = querySample.objectListField.query("FALSEPREDICATE").count().find()
+        assertEquals(0, result)
+    }
+
+    @Test
+    fun realmList_querySumEmptyList() {
+        val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+        val result = querySample.objectListField.query("FALSEPREDICATE").sum<Int>("intField").find()
+        assertEquals(0, result)
+    }
+
+    @Test
+    fun realmList_queryMaxWithNullValues() {
+        val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+        val result = querySample.objectListField.query("TRUEPREDICATE").max<Int>("nullableIntField").find()
+        assertEquals(4, result)
+    }
+
+    @Test
+    fun realmList_queryMinWithNullValues() {
+        val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+        val result = querySample.objectListField.query("TRUEPREDICATE").min<Int>("nullableIntField").find()
+        assertEquals(0, result)
+    }
+
+    @Test
+    fun realmList_querySumWithNullValues() {
+        val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+        val result = querySample.objectListField.query("TRUEPREDICATE").sum<Int>("nullableIntField").find()
+        assertEquals(6, result)
+    }
+
+    @Test
+    fun realmList_flowAddElement() {
+        runBlocking {
+            val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+            val channel = Channel<ResultsChange<*>>(capacity = 1)
+            val observer = async {
+                querySample.objectListField
+                    .query("TRUEPREDICATE").asFlow()
+                    .collect { flowList ->
+                        channel.send(flowList)
+                    }
+            }
+            channel.receive().let { resultsChange ->
+                assertIs<ResultsChange<*>>(resultsChange)
+                assertNotNull(resultsChange.list)
+                assertEquals(6, resultsChange.list.size)
+            }
+            realm.writeBlocking {
+                val queriedSample = findLatest(querySample)
+                queriedSample!!.objectListField.add(QuerySample(intField = 1))
+            }
+            channel.receive().let { resultsChange ->
+                assertIs<ResultsChange<*>>(resultsChange)
+                assertNotNull(resultsChange.list)
+                assertEquals(7, resultsChange.list.size)
+            }
+            observer.cancel()
+            channel.close()
+        }
+    }
+
+    @Test
+    fun realmList_flowDeleteElement() {
+        runBlocking {
+            val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+            val channel = Channel<ResultsChange<*>>(capacity = 1)
+            val observer = async {
+                querySample.objectListField
+                    .query("TRUEPREDICATE").asFlow()
```

Comment:
I would probably make some kind of predicate here. Just because this might skip creating a query altogether in the underlying implementation. It could be something simple as `intField >= 0`